frontend/src/components/EndoAI/Timeline.vue.js

```
@@ -1,122 +1,822 @@
+    x: 0,
+    y: 0,
+    text: ''
+});
+// Computed properties
+const duration = computed(() => props.video?.duration || 0);
+// ✅ FIX: Protected playhead calculation to prevent NaN
+const playheadPosition = computed(() => {
+    const videoDuration = duration.value;
+    const currentVideoTime = props.currentTime || 0;
+    // ✅ Guard against division by zero and invalid values
+    if (!videoDuration || videoDuration === 0 || !isFinite(videoDuration)) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)